protocol/kafka_sarama/v2/receiver.go

```diff
@@ -152,7 +152,9 @@ func (c *Consumer) startConsumerGroupLoop(cg sarama.ConsumerGroup, ctx context.C
 			return
 		// Something else happened
 		default:
-			if err == nil || err == sarama.ErrClosedClient || err == sarama.ErrClosedConsumerGroup {
+			if err == nil {
```

Comment:
When rebalance is about to happen and this condition is `true`, sarama will invoke `receiver.Cleanup()` which will close the incoming channel. I believe this will cause panic when new messages are consumed. I'm quite new to sarama, so I can't suggest a working solution at the moment.